src/project/context/flowr-analyzer-context.ts

```diff
@@ -60,16 +64,19 @@ export interface ReadOnlyFlowrAnalyzerContext {
  * If you are just interested in inspecting the context, you can use {@link ReadOnlyFlowrAnalyzerContext} instead (e.g., via {@link inspect}).
  */
 export class FlowrAnalyzerContext implements ReadOnlyFlowrAnalyzerContext {
-	public readonly files:  FlowrAnalyzerFilesContext;
-	public readonly deps:   FlowrAnalyzerDependenciesContext;
-	public readonly config: FlowrConfigOptions;
+	public readonly files:     FlowrAnalyzerFilesContext;
+	public readonly deps:      FlowrAnalyzerDependenciesContext;
+	public readonly config:    FlowrConfigOptions;
+	public readonly functions: FlowrAnalyzerFunctionsContext;
+
 
 	constructor(config: FlowrConfigOptions, plugins: ReadonlyMap<PluginType, readonly FlowrAnalyzerPlugin[]>) {
 		this.config = config;
 		const loadingOrder = new FlowrAnalyzerLoadingOrderContext(this, plugins.get(PluginType.LoadingOrder) as FlowrAnalyzerLoadingOrderPlugin[]);
 		this.files = new FlowrAnalyzerFilesContext(loadingOrder, (plugins.get(PluginType.ProjectDiscovery) ?? []) as FlowrAnalyzerProjectDiscoveryPlugin[],
             (plugins.get(PluginType.FileLoad) ?? []) as FlowrAnalyzerFilePlugin[]);
 		this.deps  = new FlowrAnalyzerDependenciesContext(this, (plugins.get(PluginType.DependencyIdentification) ?? []) as FlowrAnalyzerPackageVersionsPlugin[]);
+		this.functions = new FlowrAnalyzerFunctionsContext(this, (plugins.get(PluginType.DependencyIdentification) ?? []) as FlowrAnalyzerPackageVersionsPlugin[]);
```

Comment:
The functions context should not directly rely on the package versions, but maybe be a child of the analyzer dependencies context similar to the loading order